vault/logical_system.go

```diff
@@ -1015,6 +1017,24 @@ func (b *SystemBackend) handleUnmount(ctx context.Context, req *logical.Request,
 	return nil, nil
 }
 
+func validateMountPath(p string) error {
+	s := path.Clean(p)
+	if p != s {
```

Comment:
One wrinkle here is that "foo/" which is a valid input will become "foo" by clean which will be thrown as an error.

Reply:
This is caught by the tests as well. Can be fixed by retaining the "/" suffix on `s` if it was present in `p`.

Reply:
In the interest of time, in order to get this fix in sooner. I'll be merging this PR and fixing the breaking test in a separate PR.